src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,282 @@
+#include "widget/wfindonwebmenu.h"
+
+#include <QDesktopServices>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/math.h"
+#include "util/qt.h"
+#include "util/widgethelper.h"
+
+namespace {
+
+QString composeActionText(const QString& prefix, const QString& trackProperty) {
+    return prefix + QStringLiteral(" | ") + trackProperty;
+}
+
+QString composeSearchQuery(const QString& trackAlbumOrTitle, const QString& artist) {
+    return trackAlbumOrTitle + QStringLiteral(" ") + artist;
+}
+
+QString composePrefixAction(WFindOnWebMenu::TrackSearchProperties trackSearchProperty) {
+    if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Title) {
+        return QObject::tr("Title");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Artist) {
+        return QObject::tr("Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+        return QObject::tr("Title && Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndAlbum) {
+        return QObject::tr("Album && Artist");
+    } else {
+        return QObject::tr("Album");
+    }
+}
+
+QString searchUrlSoundCloudArtist = QStringLiteral("https://soundcloud.com/search/people?");
+
+QString searchUrlSoundCloudTitle = QStringLiteral("https://soundcloud.com/search/sounds?");
+
+QString searchUrlSoundCloudAlbum = QStringLiteral("https://soundcloud.com/search/albums?");
+
+QString searchUrlDiscogsGen = QStringLiteral("https://www.discogs.com/search/?");
```

Comment:
We can refine Discogs search, by utilize the detailed search.
But it is fine to postpone it to a follow up PR

Reply:
Exactly, I was waiting for reviews before i dive into detailed Discogs search. Because there are many queries needed for Discogs, which is different than other services.